x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/perform_bulk_action_route.ts

```diff
@@ -87,32 +104,29 @@ const normalizeErrorResponse = (
 
 const buildBulkResponse = (
   response: KibanaResponseFactory,
-  fetchRulesOutcome: PromisePoolOutcome<string, RuleAlertType>,
-  bulkActionOutcome: PromisePoolOutcome<RuleAlertType, RuleAlertType | null>
+  {
+    errors = [],
+    updated = [],
+    created = [],
+    deleted = [],
+  }: {
+    errors?: BulkActionError[];
+    updated?: RuleAlertType[];
+    created?: RuleAlertType[];
+    deleted?: RuleAlertType[];
+  }
 ) => {
-  const errors = [...fetchRulesOutcome.errors, ...bulkActionOutcome.errors];
+  const succeeded = updated.length + created.length + deleted.length;
   const summary = {
     failed: errors.length,
-    succeeded: bulkActionOutcome.results.length,
-    total: bulkActionOutcome.results.length + errors.length,
+    succeeded,
+    total: succeeded + errors.length,
```

Comment:
Nit: This was a little tricky to read, since `errors` is an array, but the other variables (`succeeded`, `failed`, `total`) are counts. Might be nice to use something like `numSucceeded`, `numFailed`, etc., just to be more clear.

Reply:
added, thanks for the suggestion